src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -521,7 +562,28 @@ def sql_query(self) -> str:
         if not self.has_clause and not self.projection:
             self._sql_query = self.from_.sql_query
             return self._sql_query
-        from_clause = self.from_.sql_in_subquery
+        if self.analyzer.session._cte_optimization_enabled:
+            placeholder = f"{analyzer_utils.LEFT_PARENTHESIS}{self.from_._id}{analyzer_utils.RIGHT_PARENTHESIS}"
+            self._sql_query = self.placeholder_query.replace(
+                placeholder, self.from_.sql_in_subquery
+            )
+        else:
+            self._sql_query = self._build_query(self.from_.sql_in_subquery)
```

Comment:
This line is different from the old code `from_clause = self.from_.sql_in_subquery`. Can we avoid this change?